usr/share/rear/output/RAWDISK/Linux-i386/270_create_grub2_efi_bootloader.sh

```diff
@@ -17,27 +17,52 @@ fi
 
 LogPrint "Using Grub 2 to create an EFI bootloader"
 
-if is_true $USING_UEFI_BOOTLOADER ; then
+if is_true $USING_UEFI_BOOTLOADER && [[ -z "$SECURE_BOOT_BOOTLOADER" ]]; then
     LogPrint "TIP: You can achieve a faster EFI boot by installing syslinux for EFI on this system"
 fi
 
 RAWDISK_BOOT_EFI_STAGING_ROOT="$TMP_DIR/EFI"
-local efi_boot_directory="$RAWDISK_BOOT_EFI_STAGING_ROOT/BOOT"
-
-mkdir $v -p "$efi_boot_directory"
-StopIfError "Could not create $efi_boot_directory"
 
 # Create a Grub 2 configuration file
-cat > "$efi_boot_directory/grub.cfg" << EOF
+local new_grub_config_file="$TMP_DIR/grub.cfg"
+cat > "$new_grub_config_file" << EOF
 set timeout=0
 set default=0
-menuentry "${RAWDISK_BOOT_GRUB_MENUENTRY_TITLE:-Rescue System}" {
+menuentry "${RAWDISK_BOOT_GRUB_MENUENTRY_TITLE:-Recovery System}" {
     linux /$(basename "$KERNEL_FILE") $KERNEL_CMDLINE
     initrd /$REAR_INITRD_FILENAME
 }
 EOF
 
-# Create a Grub 2 EFI core image
-local grub_modules=( part_gpt fat normal configfile linux video all_video )
-grub-mkimage -O x86_64-efi -o "$efi_boot_directory/BOOTX64.efi" -p "/EFI/BOOT" "${grub_modules[@]}"
-StopIfError "Error occurred during grub-mkimage of $efi_boot_directory/BOOTX64.efi"
+if [[ -n "$SECURE_BOOT_BOOTLOADER" ]]; then
+    # Using Secure Boot:
+    # We use '$SECURE_BOOT_BOOTLOADER' as a pointer into the original system's EFI tree, which should consist of
+    # signed EFI executables (and possibly companion files). We cannot touch those signed executables without
+    # breaking Secure Boot and we cannot know which companion files are actually required, so we play it safe
+    # and copy the entire EFI tree as is.
+    local original_efi_root="$(findmnt --noheadings --output TARGET --target "$SECURE_BOOT_BOOTLOADER")/EFI"
+    LogPrint "Secure Boot: Using the original EFI configuration from '$original_efi_root'"
+    cp -a $v "$original_efi_root/." "$RAWDISK_BOOT_EFI_STAGING_ROOT" || Error "Could not copy EFI configuration"
+
+    # Now we look for existing Grub configuration files and overwrite those with our own configuration. Again, to
+    # be safe, we are prepared for the situation where we might find more than one grub.cfg without knowing which
+    # one is effective, so we overwrite every one.
+    find "$RAWDISK_BOOT_EFI_STAGING_ROOT" -iname grub.cfg -print -exec cp $v "$new_grub_config_file" '{}' \;
+    StopIfError "Could not copy Grub configuration"
```

Comment:
I avoid the *IfError functions because they cannot work in case of 'set -e'
cf. https://github.com/rear/rear/pull/1450#issuecomment-324888071